library/src/scala/quoted/Quotes.scala

```diff
@@ -682,6 +682,22 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
       def copy(original: Tree)(name: String, tpt: TypeTree, rhs: Option[Term]): ValDef
       def unapply(vdef: ValDef): (String, TypeTree, Option[Term])
 
+      /** Creates a block `{ val <name> = <rhs: Term>; <body(x): Term> }`
+       *
+       *  Usage:
+       *  ```
+       *  ValDef.let(owner, "x", rhs1, Flags.Lazy) { x =>
+       *    ValDef.let(x.symbol.owner, "y", rhs2, Flags.Mutable) { y =>
+       *      // use `x` and `y`
+       *    }
+       *  }
+       *  ```
+       *
+       *  @param flags extra flags to with which the symbol should be constructed. Can be `Private | Protected | Override | Deferred | Final | Param | Implicit | Lazy | Mutable | Local | ParamAccessor | Module | Package | Case | CaseAccessor | Given | Enum | JavaStatic | Synthetic | Artifact`
```

Comment:
In my opinion, after checking what can work in a `Block` and what makes sense there, the flags consist of `Final | Implicit | Lazy | Mutable | Given | Synthetic`. To make things easier, something like `private[QuotesImpl] def validValInLetFlags` with the flags above should be defined in `FlagsModule` in `QuotesImpl.scala`, like it was before with `validValFlags` (then we would use it instead of `validValFlags` in the comment below and in the QuotesImpl.scala implementation).